src/main/java/org/colonelkai/publictransit/TravelManager.java

```diff
@@ -0,0 +1,132 @@
+        return this.travelMap
+                .keySet()
+                .parallelStream()
+                .filter(
+                        travel -> travel.getPlayer().equals(uuid)
+                )
+                .findAny();
+    }
+
+    public Optional<Travel> getTravelFromPlayer(User player) {
+        return this.getTravelFromPlayer(player.getUniqueId());
+    }
+
+    private Optional<Travel> getTravelFromScheduler(Scheduler scheduler) {
+        return this.travelMap.entrySet()
+                .parallelStream()
+                .filter((e) -> e.getValue().equals(scheduler))
+                .map(Map.Entry::getKey)
+                .findAny();
+    }
+
+    public Scheduler getScheduler(Travel travel) {
+        return this.travelMap.get(travel);
+    }
+
+    public void updateScheduler(Travel travel, Scheduler newScheduler) {
+        this.travelMap.replace(travel, newScheduler);
+    }
+
+    public void stopTravel(Travel travel, boolean returnToOriginal) {
+        this.travelMap.remove(travel);
+
+        if(returnToOriginal) {
+            TranslateCore.getScheduleManager().schedule()
+                    .setDelay(5)
+                    .setDelayUnit(TimeUnit.MINECRAFT_TICKS)
+                    .setRunner(scheduler -> {
+                        Optional<LivePlayer> optionalLivePlayer
+                                = TranslateCore.getServer().getOnlinePlayers()
+                                .parallelStream()
+                                .filter(p -> p.getUniqueId().equals(travel.getPlayer()))
+                                .findAny();
+                        if(optionalLivePlayer.isEmpty()) {
+                            return;
+                        }
+                        LivePlayer player = optionalLivePlayer.get();
+
+                        player.setPosition(travel.getStartingNode().getPosition());
+                        scheduler.cancel();
+                    })
+                    .buildRepeating(PublicTransit.getPlugin());
+        }
+    }
+
+    // The Scheduler Loop (So cool it'll make your head spin! It's also probably bad code.)
+    /*
+    enter loop by saying 'startTravel'.
+
+    'startTravel' calls 'scheduleNextTravel'
+
+    'scheduleNextTravel' does:
+    - schedule a task to be run (consumer) that will do the moving to next node
+    when the time comes, which is the current node's time
+    - set runAfter so that it itself will be called after the consumer is done.
+
+    During this, consumer checks to see if it has been removed from this list here, and
+    terminates if so, not letting the runAfter schedule another scheduler to schedule. what? yeah.
+
+    something like that.
+     */
+
+    private Scheduler scheduleNextTravel(Travel travel) {
+        return TranslateCore.getScheduleManager().schedule()
+                .setDelay(travel.getCurrentNode().getTime())
+                .setDelayUnit(TimeUnit.SECONDS)
+                // RUNNER
```

Comment:
Is this comment really needed

hey look, this is the runner which is called with ``setRunner`` 
oh s*** really? couldnt have guessed

Reply:
I've added them to clearly demarkate both Lambda Functions, to see them separately clearly. It's for my dum brain.